include/sound/sof/channel_map.h

```diff
@@ -0,0 +1,65 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Marcin Maka <marcin.maka@linux.intel.com>
+ *         Slawomir Blauciak <slawomir.blauciak@linux.intel.com>
+ */
+
+#ifndef __IPC_CHANNEL_MAP_H__
+#define __IPC_CHANNEL_MAP_H__
+
+#include <ipc/header.h>
+#include <stdint.h>
+
+/**
+ * \brief Channel map, specifies transformation of one-to-many or many-to-one.
```

Comment:
how do we handle M-N mappings? You would have N mappings dealing with M channels?

Reply:
Sorry, I'm not sure if I understand what you mean by M and N

Reply:
@slawblauciak I think that means many-to-many.

Reply:
We don't have a scenario like that to my knowledge.

Reply:
The one-to-many and many-to-one configurations refer to the number of streams. So, the way 5.1 to stereo would be handled would depend on whether it's a single stream or not. If it is, it could be mapped using either configuration I believe.

Reply:
ok, then adding a comment would help. If I didn't get it it's likely others would miss this point as well.

Reply:
I will be removing the doxygen comments though, just spoke with @mmaka1.
He said something about some other doc system being used for the kernel.

Reply:
yeah, the kernel uses the REST stuff, same as SOF docs.
I don't know if you really need to remove these comments, there are already present e.g. in header.h. I'd prefer it if we avoided changes really and used the same __packed shortcut.

Reply:
@plbossart @slawblauciak actually, I think, those comments should be in "kernel-doc" format https://www.kernel.org/doc/html/v5.2/doc-guide/kernel-doc.html